magma_ff/create_metawflrun_handler.py

```diff
@@ -0,0 +1,358 @@
+
+    @cached_property
+    def retrieved_meta_workflow_handler(self):
+        """
+        Acquire fields from associated MetaWorkflow Handler needed to create the Run Handler
+        """
+        # TODO: a check to make sure it is indeed of mwf handler type? does this function exist on ff_utils?
+        # same for above associated item request
+        meta_workflow_handler = make_embed_request(
+            self.meta_workflow_handler_identifier,
+            self.META_WORKFLOW_HANDLER_FIELDS,
+            self.auth_key,
+            single_item=True,
+        )
+        if meta_workflow_handler:
+            raise MetaWorkflowRunHandlerCreationError(
+                f"No MetaWorkflow Handler found for given 'meta_workflow_handler' identifier: {self.meta_workflow_handler_identifier}"
+            )
+        return meta_workflow_handler
+
+    @cached_property # made cached because topological sort can return different valid results
+    def meta_workflow_handler_instance(self):
+        """
+        Creates MetaWorkflowHandler object.
+        This induces topological sort of steps and validation of attributes.
+        """
+        return MetaWorkflowHandler(self.retrieved_meta_workflow_handler)
+
+    @property
+    def ordered_meta_workflow_names(self):
+        """
+        Initializes a MetaWorkflowHandler object, which topologically sorts its
+        MetaWorkflow steps and contains attribute of these steps in order,
+        the ordered_meta_workflows array.
+
+        :returns: ordered_meta_workflows attribute
+        :rtype: list[str]
+        """
+        # Extract the ordered list of MetaWorkflow names
+        return  getattr(
+                self.meta_workflow_handler_instance, ORDERED_META_WORKFLOWS
+            )
+
+    @property
+    def meta_workflow_steps(self):
+        """
+        Initializes a MetaWorkflowHandler object, which topologically sorts its
+        MetaWorkflow steps and contains attribute of these steps in order,
+        the meta_workflows array.
+
+        :returns: meta_workflows attribute
+        :rtype: list[dict]
+        """
+        # Create MetaWorkflowHandler object
+        # This ensures all necessary attrs are present in the following Run Handler creation
+        # and that MetaWorkflow Steps are topologically sorted
+        associated_meta_workflow_handler_object = MetaWorkflowHandler(
+            self.retrieved_meta_workflow_handler
+        )
+
+        # Extract the ordered list of MetaWorkflow names
+        return  getattr(
+                associated_meta_workflow_handler_object, META_WORKFLOWS
+            )
+
+    @property
+    def get_project(self):
+        """Retrieves project attribute from the associated item."""
+        return self.retrieved_associated_item.get(PROJECT)
+
+    @property
+    def get_institution(self):
+        """Retrieves institution attribute from the associated item."""
+        return self.retrieved_associated_item.get(INSTITUTION)
```

Comment:
As a personal preference, consider using `get` in a method name for non-attribute/property methods, and using attribute-like names for properties (they live somewhere in-between, but I think of them more like attributes than methods).